### PR TITLE
Throw error when there are pending migrations

### PR DESCRIPTION
**Pull request by mbareta**

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ const capitalize = require('to-case').capital;
 const pkg = require('../package.json');
 const { promisify } = require('util');
 const sequelize = require('sequelize');
+const umzug = require('./shared/database/umzug');
```
Comment (vladimyr) [inline server/index.js]: ⚠️ Loading this loads `./shared/database/index.js` too which in fact loads all our models before `sequelize.Promise` gets configured.

File: server/index.js [hunk 2]
```diff
@@ -20,6 +21,13 @@ const logger = require('./shared/logger');
 const runApp = promisify(app.listen.bind(app));
 
 database.initialize()
+  .then(() => umzug.pending())
+  .then(migrations => {
+    if (migrations.length) {
+      return bluebird.reject(new Error('There are pending migrations.'));
+    }
+    return bluebird.resolve();
```
Comment (vladimyr) [inline server/index.js]: You are inside promise chain anyway, anything you return (_or do not return_) will get wrapped inside promise 😉

File: server/shared/database/umzug.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  // The logging function.
+  // A function that gets executed everytime migrations start and have ended.
+  logging: false,
+  migrations: {
+    params: [db.sequelize.getQueryInterface(), db.sequelize.constructor, db],
+    path: './server/shared/database/migrations'
+  }
+});
+
+module.exports = umzug;
```
Comment (vladimyr) [inline server/shared/database/umzug.js]: Personally I think this is good candidate for creating factory that'll receive `db` module as param 🤔

Comment (mbareta) [reply server/shared/database/umzug.js]: 👌

**Comment by mbareta** [comment]
@vladimyr please another pass.

File: server/shared/database/umzug.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const Umzug = require('umzug');
+
+const umzug = db => new Umzug({
```
Comment (vladimyr) [inline server/shared/database/umzug.js]:
```suggestion
module.exports = db => new Umzug({
```

**Comment by underscope** [comment]
@kjuej Please test it

**Comment by kjuej** [comment]
@mbareta @underscope Seems to be working as expected.

**Pull request merged**